File: php/src/Model/CreateJoinLinkResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A new join link plus its URL. The URL is returned only here — a lost link is re-created, never recovered.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class CreateJoinLinkResult
{
    public function __construct(
        public readonly JoinLink $link,
        public readonly string $joinUrl,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            link: JoinLink::fromArray($data['link'] ?? []),
            joinUrl: (string) ($data['joinUrl'] ?? ''),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/CreateJoinLinkResult.java
package io.fivexer.sdk.model;

/**
 * A new join link plus its URL. The URL is returned only here — a lost link is re-created, never recovered.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class CreateJoinLinkResult {
    private JoinLink link;
    private String joinUrl;

    public JoinLink getLink() { return link; }
    public String getJoinUrl() { return joinUrl; }
}
